### PR TITLE
add worker logic

### PR DESCRIPTION
**Pull request by nousr**
add device in warm up log

fix warmup device

fix slurm cache

small fix in load clip

update readme

### CODE REVIEW

File: tests/test_clip_inference/test_mapper.py [hunk 1]
```diff
@@ -5,10 +5,11 @@
 
 from clip_retrieval.clip_inference.mapper import ClipMapper
 
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
```
Comment (rom1504) [inline tests/test_clip_inference/test_mapper.py:8]: let's put that back in test_mapper

File: tests/test_clip_inference/test_main.py [hunk 2]
```diff
@@ -4,11 +4,13 @@
 import tempfile
 from clip_retrieval.clip_inference.main import main
 
+os.environ["CUDA_VISIBLE_DEVICES"] = ""
```
Comment (rom1504) [inline tests/test_clip_inference/test_main.py:7]: let's put that back in test_main

File: tests/test_clip_inference/test_distributor.py [hunk 3]
```diff
@@ -1,79 +1,36 @@
...
-                sampler,
-                preprocess,
-                folder,
-                batch_size,
-                num_prepro_workers,
-                enable_text=False,
-                enable_image=True,
-                enable_metadata=False,
-            )
+CACHE_PATH = None
```
Comment (rom1504) [inline tests/test_clip_inference/test_distributor.py:10]: remove this

File: README.md [hunk 4]
```diff
@@ -176,6 +176,33 @@ clip_inference turn a set of text+image into clip embeddings
...
+### Inference Worker
+
+If you wish to have more control over how inference is run, you can create and call workers directly using `clip-retrieval inference.worker`
+
+Example Usage:
+
+```bash
+clip-retrieval inference.worker \
+--tasks="[0]" \
+--input_dataset="pipe:aws s3 cp --quiet s3://s-datasets/laion5b/laion2B-data/{000000..000010}.tar -" \
```
Comment (rom1504) [inline README.md:188]: can you put a more dummy path like `"input/folder/{000000..000100}.tar"` ?

File: README.md [hunk 5]
```diff
@@ -176,6 +176,33 @@ clip_inference turn a set of text+image into clip embeddings
...
+--output_folder="s3://my-data/worker-test" \
```
Comment (rom1504) [inline README.md:189]: let's use output_folder as example path

File: README.md [hunk 6]
```diff
@@ -176,6 +176,33 @@ clip_inference turn a set of text+image into clip embeddings
...
+--input_format="webdataset" \
+--output_partition_count="1"
+```
+
+Doing so will invoke a single worker that can be instructed to focus on a specific subset of the `input_dataset`. That worker will sequentially process the `tasks` passed to it. Here, `tasks` is a lists of `partition_id`'s that this worker will be responsible for.
```
Comment (rom1504) [inline README.md:194]: let's document these arguments in the same format as above

and maybe say that the total number of partition is number_samples / wds_number_file_per_input_file

File: tests/test_clip_inference/test_worker.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import os
+import numpy as np
+
+import tempfile
+from clip_retrieval.clip_inference.worker import worker
+
+
+def test_main():
```
Comment (rom1504) [inline tests/test_clip_inference/test_worker.py:8]: test_worker

**Comment by nousr** [comment]
all tests passing, let me know if you are satisfied with the README changes

**Comment by rom1504** [comment]
Great, thanks

**Pull request merged**